Sources/Architecture/X86/SoftwareBreakpointManager.cpp

```diff
@@ -34,9 +34,11 @@ void SoftwareBreakpointManager::clear() {
 }
 
 ErrorCode SoftwareBreakpointManager::add(Address const &address, Type type,
-                                         size_t size) {
+                                         size_t size, Mode mode) {
   DS2ASSERT(size == 0 || size == 1);
-  return super::add(address, type, size);
+  DS2ASSERT(mode != kModeExec);
```

Comment:
That's wrong, you want `DS2ASSERT(mode == kModeExec);`. I'll fix it before pushing this patch.